engine/src/main/scala/cromwell/webservice/routes/wes/RunListResponse.scala

```diff
@@ -0,0 +1,19 @@
+package cromwell.webservice.routes.wes
+
+import cromwell.services.metadata.MetadataService
+import cromwell.services.metadata.MetadataService.{WorkflowQueryFailure, WorkflowQuerySuccess}
+import cromwell.webservice.routes.wes.WesState.fromStatusString
+
+
+case class RunListResponse(runs: List[WesRunStatus], next_page_token: String)
+
+object RunListResponse {
+  def fromMetadataQueryResponse(response: MetadataService.MetadataQueryResponse): RunListResponse = {
+    response match {
+      case w: WorkflowQuerySuccess =>
+        val runs = w.response.results.toList.map(x => WesRunStatus(x.id, fromStatusString(x.status)))
+        RunListResponse(runs, "Not Yet Implemented")
+      case _: WorkflowQueryFailure => ??? // FIXME: What do we return when we fail to get a Workflow Query?
```

Comment:
If `fromMetadataQueryResponse` sometimes fails, it's better to encode that in the return type than throw an undeclared exception mid-code.

Some options to think about:
1. The return type (`RunListResponse`) could be expanded to be either a success or failure type response
1. The return type could be a level higher in the response hierarchy (eg is there a generic `Response` type)? (I guess this is just another version of (1)...?)
1. The return type could be a `ErrorOr[RunListResponse]`

Reply:
In `WesResponse.scala` there are several different case classes that extend the sealed trait `WesResponse extends Product with Serializable`. In here, the lives `WesResponseRunList` which is similar to `RunListResponse`, but it does not require the second parameter of a `next_page_token`. As far as I've found, this is the only hierarchy structure for this particular class.

Currently exploring the `ErrorOr[RunListResponse]` suggestion...